### DISCUSSION TITLE
Is there an example for supporting custom types for parse_many / document_stream

### DISCUSSION THREAD

**Discussion by pointerpower** [simdjson/simdjson#2155] [Q&A]
First, thanks for the very thorough documentation, I've been able to get up and running very quickly.

I followed the instructions for [Adding support for custom types](https://github.com/simdjson/simdjson/blob/master/doc/basics.md#adding-support-for-custom-types) to specialize simdjson::ondemand::document::get() and  simdjson::ondemand::value::get(), that worked like a charm.

However, I can't get it to work using parse_many as described here: [parse_many.md](https://github.com/simdjson/simdjson/blob/master/doc/parse_many.md#tracking-your-position). 

Specifically, once I've gotten a single doc from the stream: 
```
    ...
    for(; i != stream.end(); ++i) {
        auto streamDoc = *i;

       CustomType(streamDoc) // compiler error since doc is not of type simdjson::ondemand::document or simdjson::ondemand::value
```

Is there a way to get a simdjson::ondemand::document from streamDoc?  Do I need to specialize a different get() for streamDoc?  Please provide an example if this is possible.

**Comment by lemire** [comment]
That's a reasonable request. The type is `document_reference` but you are right. We need to have an example. We will do so.

Thanks.
